Add config docs. (#979)
earthly/earthly: docs/earthly-command/earthly-command.md

Restarts the buildkit daemon and completely resets the cache directory.

## earthly config

#### Synopsis

```
earthly [options] config [key] [value]
```

#### Description

> Enables modifying, adding, and updating values in `~/.earthly/config.yml`. It does its best to preserve formatting and existing comments. You can also set complex values using YAML literals.

Reviewer: I wonder if we can make this more concise like this?
```suggestion
Manipulates values in `~/.earthly/config.yml`. The `value` parameter can be a string or YAML literals.
```

Author: Its not just strings, its literally only YAML literals. Which include bare numbers and boolean values too. I can make the first word manipulates.

Author: I shortened it some?

#### Options

> #### `--help`

Reviewer 2: The options should have 5 `#`s
